src/__tests__/subscription.test.ts

```diff
@@ -0,0 +1,1218 @@
+      })
+    });
+
+    const pubsub = new EventEmitter();
+
+    await expectPromiseToThrow(
+      () => createSubscription(pubsub, invalidEmailSchema),
+      'Subscription field must return Async Iterable. Received: "test".'
+    );
+  });
+
+  it("resolves to an error for subscription resolver errors", async () => {
+    // Returning an error
+    const subscriptionReturningErrorSchema = emailSchemaWithResolvers(
+      () => new Error("test error")
+    );
+    await testReportsError(subscriptionReturningErrorSchema);
+
+    // Throwing an error
+    const subscriptionThrowingErrorSchema = emailSchemaWithResolvers(() => {
+      throw new Error("test error");
+    });
+    await testReportsError(subscriptionThrowingErrorSchema);
+
+    // Resolving to an error
+    const subscriptionResolvingErrorSchema = emailSchemaWithResolvers(() =>
+      Promise.resolve(new Error("test error"))
+    );
+    await testReportsError(subscriptionResolvingErrorSchema);
+
+    // Rejecting with an error
+    const subscriptionRejectingErrorSchema = emailSchemaWithResolvers(() =>
+      Promise.reject(new Error("test error"))
+    );
+    await testReportsError(subscriptionRejectingErrorSchema);
+
+    async function testReportsError(schema: GraphQLSchema) {
+      // Promise<AsyncIterable<ExecutionResult> | ExecutionResult>
+      const result = await subscribe({
+        schema,
+        document: parse(`
+          subscription {
+            importantEmail
+          }
+        `)
+      });
+
+      expect(result).toEqual({
+        errors: [
+          {
+            message: "test error",
+            locations: [{ line: 3, column: 13 }],
+            path: ["importantEmail"]
+          }
+        ]
+      });
+    }
+  });
+
+  it("resolves to an error for source event stream resolver errors", async () => {
+    // Returning an error
+    const subscriptionReturningErrorSchema = emailSchemaWithResolvers(
+      () => new Error("test error")
+    );
+    await testReportsError(subscriptionReturningErrorSchema);
+
+    // Throwing an error
+    const subscriptionThrowingErrorSchema = emailSchemaWithResolvers(() => {
+      throw new Error("test error");
+    });
+    await testReportsError(subscriptionThrowingErrorSchema);
+
+    // Resolving to an error
+    const subscriptionResolvingErrorSchema = emailSchemaWithResolvers(() =>
+      Promise.resolve(new Error("test error"))
+    );
+    await testReportsError(subscriptionResolvingErrorSchema);
+
+    // Rejecting with an error
+    const subscriptionRejectingErrorSchema = emailSchemaWithResolvers(() =>
+      Promise.reject(new Error("test error"))
+    );
+    await testReportsError(subscriptionRejectingErrorSchema);
+
+    async function testReportsError(schema: GraphQLSchema) {
+      // Promise<AsyncIterable<ExecutionResult> | ExecutionResult>
+      const result = await subscribe({
+        schema,
+        document: parse(`
+        subscription {
+          importantEmail
+        }
```

Comment:
I would move this block two spaces forward if you want to keep the test values similar to graphql-js.

Reply:
That seems to do the trick. Thanks!